### PR TITLE
Add documentation for disk based eviction

### PR DESCRIPTION
**Pull request by derekwaynecarr**
This documents the new features in OCP 3.4 for detecting and responding to local disk pressure.

@sjenning - can you please provide a review?

FYI @sdodson

### CODE REVIEW

File: admin_guide/out_of_resource_handling.adoc [hunk 1]
```diff
@@ -184,12 +210,44 @@ eviction signal.
...
 |===
 
 When the above is set the node continues to report node status updates at the
 frequency specified by the `node-status-update-frequency` argument, which
 defaults to *10s*.
 
+[[out-of-resource-reclaiming-node-level-resources]]
+=== Reclaiming node level resources
+
+If an eviction threshold has been met and the grace period has passed, the node
```
Comment (sjenning) [inline admin_guide/out_of_resource_handling.adoc]: do we want to clarify that the grace period only applies to soft thresholds?

File: admin_guide/out_of_resource_handling.adoc [hunk 2]
```diff
@@ -184,12 +210,44 @@ eviction signal.
...
+will initiate the process of reclaiming the pressured resource until it has
+observed the signal has gone below its defined threshold.
```
Comment (sjenning) [inline admin_guide/out_of_resource_handling.adoc]: might want to add that the node is unschedulable until the pressure is resolved.

File: admin_guide/out_of_resource_handling.adoc [hunk 3]
```diff
@@ -220,6 +278,25 @@ If the latter, the node evicts a `Guaranteed` pod that least impacts node
 stability and limits the impact of the unexpected consumption to other
 `Guaranteed` pods.
 
+Local disk is a `BestEffort` resource.  If necessary, the node will evict
+pods one at a time to reclaim disk when `DiskPressure` is encountered.  The node
```
Comment (sjenning) [inline admin_guide/out_of_resource_handling.adoc]: this is currently not the case.  disk eviction currently doesn't work as evicted pods don't have their disk-based volumes cleaned up.  Remedied by https://github.com/kubernetes/kubernetes/pull/37228

**Comment by sjenning** [comment]
@derekwaynecarr That's all I have.  Looks good!

File: admin_guide/out_of_resource_handling.adoc [hunk 4]
```diff
@@ -184,12 +210,44 @@ eviction signal.
...
+==== With Imagefs
+
+If `nodefs` filesystem has met eviction thresholds, the node will free up disk space in the following order:
+
+* Delete dead pods/containers
+
+If `imagefs` filesystem has met eviction thresholds, the node will free up disk space in the following order:
+
+* Delete all unused images
+
```
Comment (sdodson) [inline admin_guide/out_of_resource_handling.adoc]: Does this work for devicemapper thinpools?

**Comment by ahardin-rh** [comment]
I added a to_followup label, indicating that the docs team will apply minor follow-up edits in a separate PR. Other than that, is this ready to go? Thanks!

**Comment by ahardin-rh** [comment]
@derekwaynecarr can you address the questions in this PR? Thanks!

File: admin_guide/out_of_resource_handling.adoc (hunk 2 above)
Comment (derekwaynecarr) [inline admin_guide/out_of_resource_handling.adoc]: done

File: admin_guide/out_of_resource_handling.adoc (hunk 1 above)
Comment (derekwaynecarr) [inline admin_guide/out_of_resource_handling.adoc]: i just changed to "eviction criteria have been satisfied"

File: admin_guide/out_of_resource_handling.adoc (hunk 4 above)
Comment (derekwaynecarr) [inline admin_guide/out_of_resource_handling.adoc]: yes

File: admin_guide/out_of_resource_handling.adoc (hunk 3 above)
Comment (derekwaynecarr) [inline admin_guide/out_of_resource_handling.adoc]: I have the [NOTE] section below

**Comment by derekwaynecarr** [comment]
@ahardin-rh -- apologize the delay in updating to feedback, this should be good to go.

**Comment by ahardin-rh** [comment]
@derekwaynecarr Thank you!

**Comment by ahardin-rh** [comment]
[rev_history]
|xref:../admin_guide/out_of_resource_handling.adoc#admin-guide-handling-out-of-resource-errors[Handling Out of Resource Errors]
|Added information about disk-based eviction.
%

**Pull request merged**